packages/truffle-contract/test/networks.js

```diff
@@ -366,11 +366,13 @@ describe("Different networks: ", function() {
         from: ExampleTwo.defaults().from // Borrow the address from this one.
       });
 
-      NetworkExample.deployed()
-        .then(function(instance) {
-          assert.equal(NetworkExample.network_id, uri);
-          assert.equal(instance.address, json.networks[uri].address);
-          done();
+      NetworkExample.new(1)
```

Comment:
Why the extra `.new()`s?

Reply:
It appears there was a bug that allowed calling `.deployed()`on a new contract instance to set a `network_id` _before_ deployment. The docs are a bit out of date, but I _think_ that's not the correct behavior for that method.

Reply:
Oh, hm, seems a bit weird, then. I'll circle back on this next week and give it a think.